resources/scripts/ttx_diff.py

```diff
@@ -519,9 +520,60 @@ def remove_gdef_lig_caret_and_var_store(ttx: etree.ElementTree):
         if subtable is not None:
             gdef.remove(subtable)
 
+# remap the elements, and return a map of new_ix -> prev_ix
+def remap_class_def_ids(class_def: etree.ElementTree) -> Dict[int, int]:
+    current_classes = defaultdict(list)
+    for  glyph in class_def.xpath(".//ClassDef"):
+        cls = glyph.attrib["class"]
+        current_classes[cls].append(glyph.attrib['glyph'])
+
+    # match the sorting used in fonttools:
+    # https://github.com/fonttools/fonttools/blob/8a89f4f81b0068/Lib/fontTools/otlLib/builder.py#L2689
+    new_order = sorted(current_classes.values(), key=lambda s: (-len(s), s))
+    new_order_map = {name: i + 1 for (i, cls) in enumerate(new_order) for name in cls}
+    result = dict()
+    for  glyph in class_def.xpath(".//ClassDef"):
+        cls = glyph.attrib["class"]
+        new = new_order_map.get(glyph.attrib['glyph'])
+        glyph.attrib["class"] = str(new)
+        result[new] = int(cls)
+    return result
+
+
+def reorder_rules(lookup: etree.ElementTree, new_order: Dict[int, int], rule_name: str):
+    orig_order = [el for el in lookup.iterchildren(tag=rule_name)]
+    prev_el = orig_order[0].getprevious()
+    for el in orig_order:
+        lookup.remove(el)
+
+    for ix in range(len(orig_order)):
+        prev_ix = new_order.get(ix, ix)
+        el = orig_order[prev_ix]
+        el.set("index", str(ix))
+        prev_el.addnext(el)
+        prev_el = el
+
+    etree.indent(lookup, level=4)
```

Comment:
This feels overcomplicated and it's not completely obvious to me if it's necessary. If we're just sorting all the children of some parent I would expect it to be simpler. If its' more complicated than that perhaps we could add comments to explain?

Reply:
This little patch took my waaay to long, and this logic is much trickier than expected; basically the elements are part of a list of children that includes other elements that we don't want to modify, so we can't just clear all children and then append them in the correct order; we need to insert them in the correct order into the middle of the list. I'll add a comment to try and clarify this..